dbms/src/Common/TaskStatsInfoGetter.cpp

```diff
@@ -242,5 +236,33 @@ int TaskStatsInfoGetter::getCurrentTID()
     return static_cast<int>(syscall(SYS_gettid));
 }
 
+int TaskStatsInfoGetter::getDefaultTid()
+{
+    if (default_tid < 0)
+        default_tid = getCurrentTID();
+
+    return default_tid;
+}
+
+bool TaskStatsInfoGetter::checkProcessHasRequiredPermissions()
+{
+    /// 0 - wasn't checked
+    /// 1 - checked, has no permissions
+    /// 2 - checked, has permissions
+    static std::atomic<int> premissions_check_status{0};
```

Comment:
You can do check inside constructor of function-static variable and it will be thread safe automatically.